node/src/tasks/bsp_volunteer_mock.rs

```diff
@@ -0,0 +1,127 @@
+use std::str::FromStr;
+
+use log::{debug, error, info};
+use sp_core::H256;
+use sp_runtime::BoundedVec;
+use storage_hub_infra::{actor::ActorHandle, event_bus::EventHandler};
+
+use crate::services::{
+    blockchain::{
+        commands::BlockchainServiceInterface, events::NewStorageRequest,
+        handler::BlockchainService, types::ExtrinsicResult,
+    },
+    StorageHubHandler, StorageHubHandlerConfig,
+};
+
+const LOG_TARGET: &str = "bsp-volunteer-mock-task";
+
+pub struct BspVolunteerMockTask<SHC: StorageHubHandlerConfig> {
+    storage_hub_handler: StorageHubHandler<SHC>,
+}
+
+impl<SHC: StorageHubHandlerConfig> Clone for BspVolunteerMockTask<SHC> {
+    fn clone(&self) -> BspVolunteerMockTask<SHC> {
+        Self {
+            storage_hub_handler: self.storage_hub_handler.clone(),
+        }
+    }
+}
+
+impl<SHC: StorageHubHandlerConfig> BspVolunteerMockTask<SHC> {
+    pub fn new(storage_hub_handler: StorageHubHandler<SHC>) -> Self {
+        Self {
+            storage_hub_handler,
+        }
+    }
+}
+
+impl<SHC: StorageHubHandlerConfig> EventHandler<NewStorageRequest> for BspVolunteerMockTask<SHC> {
+    async fn handle_event(&self, event: NewStorageRequest) -> anyhow::Result<()> {
+        info!(
+            target: LOG_TARGET,
+            "Initiating BSP volunteer mock for location: {:?}, fingerprint: {:?}",
+            event.location,
+            event.fingerprint
+        );
+
+        // TODO: Here we would send the actual multiaddresses of this BSP.
+        let multiaddresses = BoundedVec::default();
+
+        // Build extrinsic.
+        let call =
+            storage_hub_runtime::RuntimeCall::FileSystem(pallet_file_system::Call::bsp_volunteer {
+                location: event.location,
+                fingerprint: event.fingerprint,
+                multiaddresses,
+            });
+
+        let (mut tx_watcher, tx_hash) = self
+            .storage_hub_handler
+            .blockchain
+            .send_extrinsic(call)
+            .await?;
+
+        // Wait for the transaction to be included in a block.
+        let mut block_hash = None;
+        // TODO: Consider adding a timeout.
+        while let Some(tx_result) = tx_watcher.recv().await {
+            // Parse the JSONRPC string, now that we know it is not an error.
+            let json: serde_json::Value = serde_json::from_str(&tx_result)
+                .expect("The result, if not an error, can only be a JSONRPC string; qed");
+
+            debug!(target: LOG_TARGET, "Transaction information: {:?}", json);
+
+            // Checking if the transaction is included in a block.
+            // TODO: Consider if we might want to wait for "finalized".
+            if let Some(in_block) = json["params"]["result"]["inBlock"].as_str() {
```

Comment:
Are there any other valid `result`s apart from `inBlock` that we should handle? I remember when I used PolkadotJS the submitted transaction could be included, finalized, etc, and all those were valid statuses that we had to handle, so I want to make sure that's not the case here